internal/magic/text_csv.go

```diff
@@ -23,8 +24,19 @@ func sv(in []byte, comma rune, limit uint32) bool {
 	r.LazyQuotes = true
 	r.Comment = '#'
 
-	lines, err := r.ReadAll()
-	return err == nil && r.FieldsPerRecord > 1 && len(lines) > 1
+	lines := 0
+	for {
+		_, err := r.Read()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return false
+		}
+		lines++
+	}
+
+	return r.FieldsPerRecord > 1 && lines > 1
```

Comment:
Given that this only checks for "lines > 1": The loop above could just break out, and there is no point at all to exhaust the reader (which could still be millions of lines?)

Reply:
The ReadAll() method would return error if any line in the file was not valid CSV. For this reason the loop calls Read() on each line and returns false overall if any line is not valid CSV.

Reply:
Breaking out of the loop when `lines > 1` means input like this would wrongfully pass as csv:
```
val1, val2, val3
val4, val5, val6

some text that is not related to the csv above
```